test/harden/lodash.js

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+require('../../src/setup_node_env');
+const _ = require('elasticsearch/node_modules/lodash'); // our fork has been patched already, using a non-patched version
+const test = require('tape');
+
+Object.prototype.sourceURL = '\u2028\u2029\n;global.whoops=true'; // eslint-disable-line no-extend-native
+
+test.onFinish(() => {
+  delete Object.prototype.sourceURL;
+});
+
+test('test setup ok', t => {
+  t.equal({}.sourceURL, '\u2028\u2029\n;global.whoops=true');
+  t.end();
+});
+
+test(`_.template('<%= foo %>')`, t => {
+  const output = _.template('<%= foo %>')({ foo: 'bar' });
+  t.equal(output, 'bar');
+  t.equal(global.whoops, undefined);
+  t.end();
+});
+
```

Comment:
Do you think it's worthwhile to test lodash with the [fourth `guard` parameter](https://github.com/lodash/lodash/blob/4.17.15/lodash.js#L14776-L14778) provided? In other words, testing that calling `_.template` as an iteree of `_.map` still works as expected? I only ask because there is special handling today to clear out the provided options:

https://github.com/lodash/lodash/blob/ddfd9b11a0126db2302cb70ec9973b66baec0975/lodash.js#L14776-L14778

Reply:
I do... I need to give the `guard` parameter more attention. I came across it on Friday, grew frustrated, and moved on...